tokio/src/runtime/scheduler/multi_thread/handle.rs

```diff
@@ -95,6 +95,31 @@ cfg_metrics! {
     }
 }
 
+cfg_taskdump! {
+    impl Handle {
+        pub(crate) fn dump(&self) -> crate::runtime::Dump {
+            let trace_status = &self.shared.trace_status;
+
+            // If a dump is in progress, block.
+            trace_status.start_trace_request(&self);
+
+            let result = loop {
+                if let Some(result) = trace_status.take_result() {
+                    break result;
+                } else {
+                    self.notify_all();
+                    crate::loom::thread::yield_now();
+                }
+            };
```

Comment:
Is it not possible to avoid this?

Reply:
It seemed like the simplest way to achieve "wait for the dump to finish and grab its result". I could use a channel, but that would add some complexity in managing the tx and rx ends. What should I be trying to avoid? Would a channel be better?

Reply:
The reason I commented on it is because this is a busy-loop, which is quite expensive.

Reply:
I still need to chase down some CI failures related the fix, but does the fix seem adequate?